### PR TITLE
UOE-11197: Log device ID and Model in Logger

### PR DESCRIPTION
**Pull request by pm-saurabh-narkhede**

### CODE REVIEW

File: modules/pubmatic/openwrap/device.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ func populateDeviceContext(dvc *models.DeviceCtx, device *openrtb2.Device) {
 	}
 	//this is needed in determine ifa_type parameter
 	dvc.DeviceIFA = strings.TrimSpace(device.IFA)
+	setDeviceIDAndModel(dvc, device)
```
[the remark this answers is not in the document]
Comment (pm-isha-bharti) [reply modules/pubmatic/openwrap/device.go:17]: In `updateDeviceIFADetails` we are updating device.IFA. Do we want to have this method after `updateDeviceIFADetails`?

File: modules/pubmatic/openwrap/device.go [hunk 2]
```diff
@@ -30,6 +31,38 @@ func populateDeviceContext(dvc *models.DeviceCtx, device *openrtb2.Device) {
...
+		dvc.ID = device.DPIDMD5
+		return
+	}
+	if device.MACSHA1 != "" {
+		dvc.ID = device.MACSHA1
+		return
+	}
+	if device.MACMD5 != "" {
+		dvc.ID = device.MACMD5
+	}
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/device.go:63]: In the [UOE-11197](https://inside.pubmatic.com:9443/jira/browse/UOE-11197), I can see, we have Device.Ext.SessionId as the last fallback, I cant see it here

Comment (pm-saurabh-narkhede) [reply modules/pubmatic/openwrap/device.go:63]: it is added in function `updateDeviceIFADetails` incase `dvc.ID` not set

File: modules/pubmatic/openwrap/device.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (pm-saurabh-narkhede) [reply modules/pubmatic/openwrap/device.go:17]: if `device.IFA` not present in request, we are updating `device.IFA` with `session_id` but anyway `session_id` has last priority and this is handled in `updateDeviceIFADetails`

File: modules/pubmatic/openwrap/device.go (hunk 2 above)
Comment (pm-isha-bharti) [reply modules/pubmatic/openwrap/device.go:63]: Can this be part of this method only as this is specific to deviceID and updateDeviceIFADetails is for updating device IFA?

Comment (pm-saurabh-narkhede) [reply modules/pubmatic/openwrap/device.go:60]: Yes. Removed from `updateDeviceIFADetails` and added in parent function as we need `dvc.ext` for session_id

File: modules/pubmatic/openwrap/device.go [hunk 3]
```diff
@@ -26,10 +28,39 @@ func populateDeviceContext(dvc *models.DeviceCtx, device *openrtb2.Device) {
...
+	if dvc.ID == "" {
+		dvc.ID, _ = dvc.Ext.GetSessionID()
+	}
 	//update device IFA Details
 	updateDeviceIFADetails(dvc)
 }
 
+// getDeviceID retrieves deviceID for logging purpose
+func getDeviceID(dvc *models.DeviceCtx, device *openrtb2.Device) string {
+	if dvc.DeviceIFA != "" {
```
Comment (AvinashKapre) [inline modules/pubmatic/openwrap/device.go:40]: NitPick: Can we use swith case here?

**Review by AvinashKapre — APPROVED** [review]

**Pull request merged**